lib/rules-generator.js

```diff
@@ -218,25 +251,34 @@ util.methods(Generator, {
     var i;
     var pathMethods = ['read', 'write'];
     var location = this.ensurePath(path.parts);
+    var exp;
 
-    // Write validation (if non-trivial)
-    location['.validate'] = this.getExpressionText(
-      this.partialEval(this.symbols.functions['@validator@' + path.isType].body));
-    if (location['.validate'] == 'true') {
-      delete location['.validate'];
+    // Path validation function && Type Validation
+    var terms = [];
+    if (path.methods.validate) {
+      terms.push(path.methods.validate.body);
+    }
+    terms.push(this.symbols.functions['@validator@' + path.isType].body);
+
+    exp = this.partialEval(ast.andArray(terms));
+    if (!(exp.type == 'Boolean' && exp.value == true)) {
+      location['.validate'] = this.getExpressionText(exp);
     }
 
     // Write read and write methods
     for (i = 0; i < pathMethods.length; i++) {
       var method = pathMethods[i];
       if (path.methods[method]) {
-        location['.' + method] = this.getExpressionText(path.methods[method].body);
+        exp = path.methods[method].body;
+        if (!(exp.type == 'Boolean' && exp.value == false)) {
+          location['.' + method] = this.getExpressionText(exp);
+        }
       }
     }
 
     // Write indices
     if (path.methods.index) {
-      var exp = path.methods.index.body;
+      exp = path.methods.index.body;
```

Comment:
assignment followed by reassignment is nasty. I try to aim for SSA if possible. You can use ternary to avoid an intermediate assignment. exp = path.methods.index.body.type != 'String' ?  path.methods.index.body : ast.array([path.methods.index.body]). However I think in this case you want to to push the fact that true is "True" into the AST layer, or another method before this. Rather than doing alternative path here.